scripts/irods/test/test_iget.py

```diff
@@ -123,3 +121,58 @@ def test_iget_does_not_change_replica_status__issue_5760(self):
             # Make sure the data path is real so that it will get cleaned up
             self.admin.assert_icommand(['iadmin', 'modrepl', 'logical_path', logical_path, 'replica_number', '0', 'DATA_PATH', real_data_path])
 
+    def test_single_buffer_get_with_max_single_buffer_size_of_min_or_max_value_works__issue_8373(self):
+        content = "get this"
+        logical_path = "/".join([self.user.session_collection, "get_this.txt"])
+
+        try:
+            self.user.assert_icommand(["istream", "write", logical_path], input=content)
+
+            server_config_path = paths.server_config_path()
+            with open(server_config_path) as f:
+                server_config = json.load(f)
+
+            # These buffer sizes are the minimum and maximum allowable values for this configuration.
+            for buffer_size_in_mb in [1, 2047]:
+                with self.subTest(f"buffer size: {buffer_size_in_mb}"):
+                    server_config["advanced_settings"]["maximum_size_for_single_buffer_in_megabytes"] = buffer_size_in_mb
+                    with lib.file_backed_up(server_config_path):
+                        with open(server_config_path, 'w') as f:
+                            f.write(json.dumps(server_config, sort_keys=True, indent=4, separators=(',', ': ')))
+
+                        IrodsController().reload_configuration()
+
+                        self.user.assert_icommand(["iget", logical_path, "-"], "STDOUT", content)
+
+        finally:
+            IrodsController().reload_configuration()
+
+    def test_single_buffer_get_with_max_single_buffer_size_that_overflows_returns_configuration_error__issue_8373(self):
+        content = "get this"
+        logical_path = "/".join([self.user.session_collection, "get_this.txt"])
+
+        try:
+            self.user.assert_icommand(["istream", "write", logical_path], input=content)
+
+            server_config_path = paths.server_config_path()
+
+            with open(server_config_path) as f:
+                server_config = json.load(f)
+
+            # This buffer size is 2048MB, which in bytes is 2048 * 1024 * 1024 = 2147483648. The value is stored in
+            # an int, which for our supported platforms has a max value of 2147483647 (2^31 - 1). This should cause
+            # value to overflow. The test ensures that this is handled gracefully by returning an error regarding
+            # the configuration.
+            buffer_size_in_mb = 2048
+            server_config["advanced_settings"]["maximum_size_for_single_buffer_in_megabytes"] = buffer_size_in_mb
+
+            with lib.file_backed_up(server_config_path):
+                with open(server_config_path, 'w') as f:
+                    f.write(json.dumps(server_config, sort_keys=True, indent=4, separators=(',', ': ')))
+
+                IrodsController().reload_configuration()
+
+                self.user.assert_icommand(["iget", logical_path, "-"], "STDERR", "CONFIGURATION_ERROR")
```

Comment:
Let's include the error code in the expected output too.

Reply:
This test has been removed.